packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -0,0 +1,255 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import {
+	createElement,
+	Fragment,
+	useCallback,
+	useEffect,
+	useReducer,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { closeSmall, external } from '@wordpress/icons';
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import {
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlMenu as Menu,
+	__experimentalSelectControlMenuItem as MenuItem,
+	useAsyncFilter,
+	Spinner,
+} from '@woocommerce/components';
+import { Product } from '@woocommerce/data';
+import { getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { FormattedPrice } from '../../../components/formatted-price';
+import { ProductEditorBlockEditProps } from '../../../types';
+import {
+	getLoadLinkedProductsDispatcher,
+	getRemoveLinkedProductDispatcher,
+	getSearchProductsDispatcher,
+	getSelectSearchedProductDispatcher,
+	reducer,
+} from './reducer';
+import { LinkedProductListBlockAttributes } from './types';
+
+export function getProductImageStyle( product: Product ) {
+	return product.images.length > 0
+		? {
+				backgroundImage: `url(${ product.images[ 0 ].src })`,
+		  }
+		: undefined;
+}
+
+export function Edit( {
```

Comment:
It would be good to split out the following components from this implementation file to reusable components under `src/components` that could be used by other block implementations, such as custom blocks:

- Product select
- Products list